src/instructlab/model/backends/backends.py

```diff
@@ -165,12 +173,12 @@ def get(model_path: pathlib.Path, backend: str | None) -> str:
     # If the backend was set using the --backend flag, validate it.
     else:
         logger.debug(f"Validating '{backend}' backend")
-        # TODO: keep this code logic and implement a `--force` flag to override the auto-detected backend
         # If the backend was set explicitly, but we detected the model should use a different backend, raise an error
-        # if backend != auto_detected_backend:
-        #     raise ValueError(
-        #         f"The backend '{backend}' was set explicitly, but the model was detected as '{auto_detected_backend}'."
-        #     )
+        if backend != auto_detected_backend:
+            logger.warning(
+                f"The backend '{backend}' was set explicitly, but the model was detected as '{auto_detected_backend}, reason: {auto_detected_backend_reason}'.\n"
```

Comment:
This wording seems confusing to me, could we considering rephrasing this to something along the lines of "The serving backend '{backend}' was requested explicity, but the provided model is not compatible with it"?